content/articles/flow-and-typescript-2.md

```diff
@@ -0,0 +1,108 @@
+
+As mentioned in the introduction, TypeScript can target ES5 or ES6. The advantage of targeting ES6 is that you can still use Babel to compile and benefit from hot module reload. For more information on how to do that, please read [ES6 modules with TypeScript and webpack](www.jbrantly.com/es6-modules-with-typescript-and-webpack/) and have a look at the [webpack config](https://github.com/Keats/flow-typescript/blob/master/typescript/webpack.config.js) for a practical example.
+
+
+## Actual conversion
+TypeScript use a config file named `tsconfig.json`. Here's the one currently used by the example:
+
+```ts
+{
+  "compilerOptions": {
+    "target": "es6",
+    "jsx": "react",
+    "noImplicitAny": true,
+    "sourceMap": true
+  },
+  "exclude": ["node_modules", "gulpfile.js", "webpack.config.js", "server.js"]
+}
+```
+When you are starting the conversion, you want to remove the `"noImplicitAny": true,` line since it would error if you have any missing type annotations.
+
+You then need to rename all your `.js` files to `.ts` and `.tsx`. The `.txs` extension is needed if the file contains JSX.
+
+## Some gotchas and examples
+While TypeScript intends to support most of ES6 and ES7 features, you will need to change some bits of your code. Here are the main ones I stumbled into.
+
+### Object destructuring assignment
+It doesn't work in TypeScript ([playground](http://www.typescriptlang.org/Playground#src=var%20printer%20%3D%20function%28obj%3A%20Object%29%20{%0A%09console.log%28obj%29%3B%0A}%0A%0Aconst%20hey%20%3D%20{1%3A%20null%2C%202%3A%20null}%3B%0A%0Aprinter%28{%0A%09lol%3A%20null%2C%0A%09...hey%0A}%29%3B)).
+Let's look at the `combineReducers` from redux for example:
+
+```js
+const reducer = combineReducers({
+  router: routerStateReducer,
+  ...reducers,
+});
+// will need to be changed to
+const reducer = combineReducers({
+  router: routerStateReducer,
+  lists: reducers.lists,
+  cards: reducers.cards
+});
+```
+
+### Importing default
+Importing the default from a ES6 module is different from the Babel one. For example in the case of React:
+
+```ts
+// will not work in typescript
+import React from "react";
+// will work in typescript
+import * as React from "react";
+```
+
+### Adding 3rd party definitions
+Adding type definitions for your own app is great but if you are using 3rd party libraries without definitions for them It's a bit like giving a mail to a random mail carrier and hoping it was the right one.
+
+Thankfully we have [DefinitelyTyped](https://github.com/borisyankov/DefinitelyTyped) to save us. While this doesn't have definitions for every single library you might use, it contains a fair bit.
+There is a tool called [tsd](http://definitelytyped.org/tsd/) to manage definitions but frankly, I don't use so many libraries that it's worth the hassle over manually copying files.
+
+
+### React PropTypes
+Now that you have realy typing, you can delete all of those and add interfaces for your components props and state:
```

Comment:
the next sentence doesn't make sense. "you can delete all the React `propTypes` and use the following instead" maybe?